File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/IdentityType.java
package com.coinninja.coinkeeper.model.db.enums;

public enum IdentityType {
    UNKNOWN(-1),
    PHONE(0),
    TWITTER(1);
    private final int id;

    IdentityType(int id) {
        this.id = id;
    }

    public static IdentityType from(String type) {
        if (type == null) return UNKNOWN;

        switch (type.toLowerCase()) {
            case "phone":
                return PHONE;
            case "twitter":
                return TWITTER;
            default:
                return UNKNOWN;
        }

    }

    public int getId() {
        return id;
    }

    public String asString() {
        switch (this) {
            case PHONE:
                return "phone";
            case TWITTER:
                return "twitter";
            default:
                return "unknown";
        }
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/IdentityType.kt
package com.coinninja.coinkeeper.model.db.enums

enum class IdentityType(val id: Int) {
    UNKNOWN(-1),
    PHONE(0),
    TWITTER(1);

    fun asString(): String {
        when (this) {
            PHONE -> return "phone"
            TWITTER -> return "twitter"
            else -> return "unknown"
        }
    }

    companion object {

        fun from(type: String?): IdentityType {
            if (type == null) return UNKNOWN

            when (type.toLowerCase()) {
                "phone" -> return PHONE
                "twitter" -> return TWITTER
                else -> return UNKNOWN
            }

        }
    }
}
